### PR TITLE
Finish refactoring Module

### PR DESCRIPTION
**Pull request by melvinw**

### CODE REVIEW

File: core/module.h [hunk 1]
```diff
@@ -233,6 +232,26 @@ class Module {
     return attrs_;
   }
 
+  const std::vector<Task *> tasks() const {
```
Comment (barath) [inline core/module.h]: Mark accessors as inline.

**Comment by barath** [comment]
Would you be able to rebase?

**Comment by melvinw** [comment]
Done

**Comment by barath** [comment]
You're right -- my mistake.  I forgot this was a .h file.

**Comment by melvinw** [comment]
rolled that change back

**Comment by barath** [comment]
image

File: core/module.h [hunk 2]
```diff
@@ -234,6 +233,28 @@ class Module {
...
+    return tasks_;
+  }
+
+  void set_attr_offset(size_t idx, bess::metadata::mt_offset_t offset) {
+    if (idx < bess::metadata::kMaxAttrsPerModule) {
+      attr_offsets_[idx] = offset;
+    }
+  }
+
+  const bess::metadata::mt_offset_t *all_attr_offsets() const {
```
Comment (sangjinhan) [inline core/module.h]: If it is used in the form of `all_attr_offsets()[i])` in most cases, how about using `mt_offset_t attr_offset(size_t idx)` instead?

Comment (melvinw) [inline core/module.h]: Ah, that's cleaner. Will submit another pr
